docs/tutorial_functional.rst

```diff
@@ -0,0 +1,151 @@
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_functional_etl_dag.py
+    :language: python
+    :start-after: [START main_flow]
+    :end-before: [END main_flow]
+
+That's it, we are done!
+We have invoked the Extract task, obtained the order data from there and sent it over to
+the Transform task for summarization, and then invoked the Load task with the summarized data.
+The dependencies between the tasks and the passing of data between these tasks which could be
+running on different workers on different nodes on the network is all handled by Airflow.
+
+But how?
+--------
+For experienced Airflow DAG authors, this is startlingly simple! Let's contrast this with
+how this DAG had to be written before Airflow 2.0 below:
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_etl_dag.py
+    :language: python
+    :start-after: [START tutorial]
+    :end-before: [END tutorial]
+
+All of the processing shown above is being done in the new Airflow 2.0 dag as well, but
+it is all abstracted from the DAG developer.
+
+Let's examine this in detail by looking at the Transform task in isolation since it is
+in the middle of the data pipeline. In Airflow 1.x, this task is defined as shown below:
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_etl_dag.py
+    :language: python
+    :start-after: [START transform_function]
+    :end-before: [END transform_function]
+
+As we see here, the data being processed in the Transform function is passed to it using Xcom
+variables. In turn, the summarized data from the Transform function is also placed
+into another Xcom variable which will then be used by the Load task.
+
+Contrasting that with Functional DAGs in Airflow 2.0 as shown below.
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_functional_etl_dag.py
+    :language: python
+    :start-after: [START transform]
+    :end-before: [END transform]
+
+All of the Xcom usage for data passing between these tasks is abstracted away from the DAG author
+in Airflow 2.0. However, Xcom variables are used behind the scenes and can be viewed using
+the Airflow UI as necessary for debugging or DAG monitoring.
+
+Similarly, task dependencies are automatically generated within Functional DAGs based on the
+functional invocation of tasks. In Airflow 1.x, tasks had to be explicitly created and
+dependencies specified as shown below.
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_etl_dag.py
+    :language: python
+    :start-after: [START main_flow]
+    :end-before: [END main_flow]
+
+In contrast, with Functional DAGs in Airflow 2.0, the invocation itself automatically generates
+the dependencies as shown below.
+
+.. exampleinclude:: /../airflow/example_dags/tutorial_functional_etl_dag.py
+    :language: python
+    :start-after: [START main_flow]
+    :end-before: [END main_flow]
+
+
+What's Next?
+-------------
+You have seen how simple it is to write DAGs using the Functional DAG paradigm within
```

Comment:
Would be nice to have this point to https://github.com/apache/airflow/blob/master/docs/concepts.rst#functional-dags and https://github.com/apache/airflow/blob/master/docs/concepts.rst#functional-dags more specifically. 

Also, it may be interesting to link the AIP-31 for more insight into this and potentially my talk and Jonathan's article as learn more resources? CC @turbaszek for thoughts

Reply:
+1 for linking AIP and this article for more insight:
https://medium.com/databand-ai/aip-31-airflow-functional-dag-definition-b34852a632d0

Reply:
I added the links to the AIP and to the Functional DAGs section of the Concept doc. 

I was not sure if it was ok to add a link to an external site which has a paywall (Medium) from the Apache open source docs, so did not add that.